pkg/processor/build/builder.go

```diff
@@ -336,7 +347,8 @@ func (b *Builder) resolveFunctionPath(functionPath string) (string, error) {
 
 func (b *Builder) decompressFunctionArchive(functionPath string) (string, error) {
 	// create a staging directory
-	tempDir, err := ioutil.TempDir("", "nuclio-decompress-")
+	tempDir := path.Join(b.tempDir, decompressDirName)
```

Comment:
Seems like this repeats itself in several places. I suggest creating a private function to reduce that i.e. `func (b *Builder) mkdirUnderTempDir(name string) (string, error)` and using it instead i.e. `decompressDir, err := b.mkdirUnderTempDir(decompressDirName)`